src/app.ts

```diff
@@ -13,14 +14,61 @@ interface TemperatureSummary {
   average: number
 }
 
+/* Object containing a dictionary of dictionaries */
+const sumarys: { [key: string]: { [item: string]: TemperatureSummary } } = {}
+
+/* Function that given the array of TemperatureReadings fills the sumarys object,
+  the first dictionaries with cities and the dictionary inside of the cities
+  with the dates containing the sumary of that day */
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (let read of readings) {
+    let exist = read.city in sumarys ? (read.time.toDateString() in sumarys[read.city]) ? true : false : false
```

Comment:
I would skip the ternary here and just use a logic gate

```suggestion
    let exist = read.city in sumarys && (read.time.toDateString() in sumarys[read.city])
```

Reply:
Also since you're using this same logic elsewhere I would probably pull it out into a function